python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -137,7 +137,7 @@ def locations(self):
 
     def _setLFNnamePattern(self, lfn="", namePattern=""):
 
-        if self.defaultSE != "":
+        if hasattr(self, 'defaultSE') and self.defaultSE != "":
```

Comment:
hmmm this strikes me as a failing of the `GangaObject` and `ObjectMeta` classes as something like this is perfectly possible, see the simple example below.
```python
class ObjectMeta(type):
    def __new__(meta, name, bases, dct):
        dct['__slots__'] = dct['schema'].keys()
        return super(ObjectMeta, meta).__new__(meta, name, bases, dct)

class Object(object):
    __metaclass__ = ObjectMeta
    schema = {}

    def __getattr__(self, attr):
        if attr in self.__class__.schema:
            return self.__class__.schema.get(attr)
        return super(Object, self).__getattr__(attr)

class bob(Object):
    schema = {'frank':'default'}

    def __init__(self):
        print self.frank


>>> b=bob()
default
```
__Note__: I realise that this will not properly work with multiple instances, it's a proof that these things can be pre-setup. My schema is just a `dict` and so has no notion of instances whereas a more complicated Schema object could. Of course the proper way to do this is with properties which is what I did in the stuff that I was playing with

----
OK since this is a quick fix for a problem that is annoying for users otherwise I think it should be fine but it certainly highlights to me that I need to redouble my efforts to look at the schema.

You might like to think about reducing the conditional to the smaller but equivalent (I think) version below
```diff
-        if hasattr(self, 'defaultSE') and self.defaultSE != "":
+        if getattr(self, 'defaultSE', '') != '':
```